### PR TITLE
11.1 upload img kovalets

### PR DESCRIPTION
**Pull request by AndriyKovalets**
You need test next endpoints: api/user/image (method put) for update user image. For it need to specify a local file. If you specify image file with size less 10 Mb, server return ok request, otherwise server return 'bad' status code with error message. Another endpoint is api/user/image (method get) for get user image. This endpoint return user image, or "not found" status code with error message if image no found or user not set image yet.

### CODE REVIEW

File: Provis.Core/Exeptions/FileExceptions/FileException.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace Provis.Core.Exeptions.FileExceptions
+{
+    public class FileException: Exception
+    {
+        public override string Message { get; }
```
Comment (YevheniiVovk) [inline Provis.Core/Exeptions/FileExceptions/FileException.cs]: Don't override this

File: Provis.Core/Exeptions/FileExceptions/FileException.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        public string Path { get; }
+        public FileException(string message, string path)
+        {
+            Message = message;
+            Path = path;
```
Comment (YevheniiVovk) [inline Provis.Core/Exeptions/FileExceptions/FileException.cs:22]: TODO: check what with serialization

File: Provis.Core/Interfaces/Services/IFileService.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Provis.Core.Interfaces.Services
+{
+    public interface IFileService
+    {
+        Task<string> AddFileAsync(Stream stream, string folderName, string fileName);
+        Task<DownloadFile> GetFileAsync(string path);
+        Task DeleteFileAsync(string path);
```
Comment (YevheniiVovk) [inline Provis.Core/Interfaces/Services/IFileService.cs:11]: Please add method for create directory

File: Provis.Core/Services/UserService.cs [hunk 4]
```diff
@@ -97,5 +112,32 @@ public async Task<UserActiveInviteDTO> IsActiveInviteAsync(string userId)
 
             return userActiveInviteDTO;
         }
+
+        public async Task UpdateUserImageAsync(IFormFile img, string userId)
+        {
+            var user = await _userManager.FindByIdAsync(userId);
+
+            string newPath = await _fileService.AddFileAsync(img.OpenReadStream(), _imageSettings.Value.Path, img.FileName);
```
Comment (YevheniiVovk) [inline Provis.Core/Services/UserService.cs:117]: Use async methods for work with streams

Comment (AndriyKovalets) [reply Provis.Core/Services/UserService.cs:117]: There isn't async method for openReadStream

**Review by herakros — APPROVED** [review]

**Review by YevheniiVovk — APPROVED** [review]

**Pull request merged**